### PR TITLE
Remove all cow.mods usages outside of cow type

### PR DESCRIPTION
**Pull request by algorandskiy**
## Summary

`cow.mods` was accessed by logic ledger and its tests. This PR fixes it and now all `mods` accessed done only by `cow`'s methods.

## Test Plan

Update existing tests, add new tests for newly added methods to preserve coverage level.

### CODE REVIEW

File: ledger/appcow_test.go [hunk 1]
```diff
@@ -87,6 +89,26 @@ func (ml *emptyLedger) compactCertLast() basics.Round {
 	return basics.Round(0)
 }
 
+type modsData struct {
```
Comment (algorandskiy) [inline ledger/appcow_test.go:92]: moved from applications_test.go

File: ledger/applications_test.go [hunk 2]
```diff
@@ -203,69 +257,21 @@ func TestLogicLedgerGetKey(t *testing.T) {
 	a.False(ok)
 	a.Contains(err.Error(), fmt.Sprintf("app %d does not exist", assetIdx))
 
-	c.sdeltas = map[basics.Address]map[storagePtr]*storageDelta{
```
Comment (algorandskiy) [inline ledger/applications_test.go:206]: almost completely moved to appcow_test.go, as similar blocks below.

**Review by tsachiherman — COMMENTED** [review]
Looks good to me.

**Review by brianolson — APPROVED** [review]
LGTM; I pretty loosely skimmed the tests, that's a lot of test for two trivial getter functions added.

**Pull request merged**